src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -287,6 +287,11 @@ public void run(Timeout timeout) {
     @Override
     protected Position handlePosition(Position position) {
 
+        Device device = identityManager.getById(position.getDeviceId());
```

Comment:
You should use attribute lookup method instead.

Reply:
Added attribute lookup here.